sdk/identity/identity/src/credentials/managedIdentityCredential/arcMsi.ts

```diff
@@ -96,6 +96,7 @@ export const arcMsi: MSI = {
     }
 
     const requestOptions = {
```

Comment:
The same `requestOptions` gets re-used to get the token as well. Shouldnt we set allowInsecureConnection to false for that?

Reply:
Is there any test to check that the final call to get token would be using the secure connection regardless of the input?

Reply:
Well, if you mean the final call to getToken should use the insecure connection (this is what we want, because this is always a localhost request), then yes, I’ve changed the Arc tests to use `http` endpoints only, which verifies the fix, and verifies that the getRequest happens over `http` without troubles. You can see the change here: https://github.com/Azure/azure-sdk-for-js/pull/17362/files#diff-521c1ea4a25403c60a5d3120647f8a73fd7c9cb574a1b985847416d60e92d363R315-R316

Reply:
Oh sorry, my mistake. I read your earlier comment wrong. I assumed that the final call via the msiGenericGetToken method was meant to be "safe"

Reply:
my bad! yeah it’s generally not safe if it comes from an environment variable defined by Azure. I believe Azure relies on the security of the network itself rather of the security of the protocol in these cases.